frontend/README.md

```diff
@@ -1,36 +1,73 @@
-This is a [Next.js](https://nextjs.org) project bootstrapped with [`create-next-app`](https://nextjs.org/docs/app/api-reference/cli/create-next-app).
+# Frontend folder structure
 
-## Getting Started
+> **Note:** We are using Next.js App Router, not Page Router.
 
-First, run the development server:
+```
+frontend/
+  ├── app/
+  │   ├── layout.tsx            // Layout component
+  │   ├── page.tsx              // Root page
+  │   ├── error.tsx             // Error boundary component 
+  │   ├── not-found.tsx         // 404 page
+  │   ├── about/
+  │   │   └── page.tsx          // About page
+  │   ├── contact/
+  │   │   └── page.tsx          // Contact page
+  │   └── [slug]/
+  │       └── page.tsx          // Dynamic route page
+  ├── components/               // Reusable UI components
+  │   ├── Header.tsx
+  │   ├── Footer.tsx
+  │   └── Button.tsx
+  ├── lib/                      // Library code and utilities
+  │   ├── api.ts
+  │   ├── fetcher.ts
+  │   └── utils.ts
+  ├── hooks/                    // Custom React hooks
+  │   ├── useAuth.ts
+  │   ├── useFetch.ts
+  │   └── useForm.ts
+  ├── styles/                   // CSS or styled-components
+  │   ├── globals.css
+  │   └── theme.css
+  ├── public/                   // Static assets (e.g., images)
+  │   └── images/
+  ├── types/                    // TypeScript types and interfaces
+  │   ├── index.d.ts
+  │   └── models.ts
+  └── next.config.js            // Next.js configuration
 
-```bash
-npm run dev
-# or
-yarn dev
-# or
-pnpm dev
-# or
-bun dev
 ```
 
-Open [http://localhost:3000](http://localhost:3000) with your browser to see the result.
+## 1. Build the Docker image
 
-You can start editing the page by modifying `app/page.tsx`. The page auto-updates as you edit the file.
+Ensure you are in the frontend directory
 
-This project uses [`next/font`](https://nextjs.org/docs/app/building-your-application/optimizing/fonts) to automatically optimize and load [Geist](https://vercel.com/font), a new font family for Vercel.
+```sh
+docker build -t frontend .
+```
 
-## Learn More
+## 2. Run the container
 
-To learn more about Next.js, take a look at the following resources:
+macOS users:
 
-- [Next.js Documentation](https://nextjs.org/docs) - learn about Next.js features and API.
-- [Learn Next.js](https://nextjs.org/learn) - an interactive Next.js tutorial.
+```sh
+docker run --name my-app -p 3000:3000 -v $(pwd):/app -d frontend
```

Comment:
I face issues using this command without `$(pwd)` being put in quotes. This change worked for me though:
```sh
docker run --name my-app -p 3000:3000 -v "$(pwd)":/app -d frontend
```